doc/generic/pgf/text-en/pgfmanual-en-dv-axes.tex

```diff
@@ -2709,6 +2714,8 @@ \subsubsection{Creating the Axes}
 axes or ticks, but we will see the correct scaling of the attributes. Let us
 first define a data group:
 %
+% TODOsp: codeexamples: this is needed as `pre` for the following examples
+%         (thus, `setup code` doesn't work here)
```

Comment:
This I did by intention. Otherwise I have to add to each and every `codeexample` the library for e.g. the library files. The only thing which of course doesn't work now is that the `setup code` isn't printed in the manual.

Of course, if you are not able to modify the `setup code` accordingly then I will adapt the `codeexample`s.

If you can resolve the previously mentioned issue regarding the `setup code`, maybe you could add another key that can be used to store stuff needed for the following `codeexample`s but now is an equivalent for `pre`?

Reply:
`setup code` is *not* for `\usetikzlibrary`.  These have to be given in the `preamble` option because we want them printed in the manual.  That is what issue #640 is all about.